src/promptflow/tests/sdk_cli_azure_test/recording_utilities/utils.py

```diff
@@ -202,7 +202,7 @@ def sanitize_pfs_body(body: str) -> str:
 
 
 def sanitize_email(value: str) -> str:
-    return re.sub(r"([\w\.-]+)@([\w\.-]+)", r"{}@\2".format(SanitizedValues.EMAIL_USERNAME), value)
+    return re.sub(r"([\w\.-]+)@(microsoft.com)", r"{}@\2".format(SanitizedValues.EMAIL_USERNAME), value)
```

Comment:
Why changed this?

Reply:
during this PR, I find that this will mis-sanitize `@tool` in some cases, so narrow this condition.